### ISSUE TITLE
Transfer issue

### ISSUE THREAD

**Issue by goodwinpress** [davidsneighbour/colossus-wp-options-framework#109]
When user transfer site from one domain to another (all files and database, manually or using Duplicator), all settings in options page are vanishing. They're  saved in database but not displayed on the options page.

**Comment by davidsneighbour** [comment]
It would be good if you could provide the code that sets up your options. In general, TF is saving serialized options as `namespace_options` in the options table of wordpress, so I would expect it to work if that option is copied over and the code is identical. 

The namespace is built from the `name` parameter of your admin panel/tab.

**Comment by goodwinpress** [comment]
Thank you. If I get you right, there are some excerpts.

From settings page:

`add_action( 'tf_create_options', 'gpress_create_options' );
function gpress_create_options() {
$titan = TitanFramework::getInstance( 'gpress' );
 
$adminPanel = $titan->createAdminPanel( array(
  'name' => 'GP Medic',
 ) );

$normalPanel = $adminPanel->createAdminPanel( array(
    'name' => 'Общие настройки',
    'id'=> 'general-options',
) );
 
  $normalPanel->createOption( array(
    'name' => 'Фавикон для сайта - включить',
    'id' => 'fav-loc',
    'type' => 'checkbox',
    'default' => false,
    'desc' => 'Поставьте галочку, если хотите включить фавикон',
 ) );
 
$normalPanel->createOption( array(
    'name' => 'Фавикон для сайта',
    'id' => 'favicon',
    'type' => 'upload',
    'desc' => 'Создайте изображение с нужным рисунком,  сохраните в формате png  и загрузите его здесь.'
) );

$normalPanel->createOption( array(
    'name' => 'Заголовок сайта',
    'id' => 'site-title',
    'type' => 'select',
    'desc' => 'Выберите, что выводить - текст или логотип. <br />Если выбран текст, то в шапке будет выводиться заголовок, заданный вами в <a target="_blank" href="options-general.php">настройках сайта</a>',
    'options' => array(
        '1' => 'Текст',
        '2' => 'Логотип',
    ),
    'default' => '1',
) );

$normalPanel->createOption( array(
    'name' => 'Если выбран логотип',
    'id' => 'themelogo',
    'type' => 'upload',
    'desc' => 'Если выбран логотип - загрузите его здесь. <br />Рекомендуемый размер логотипа - 300х70 пикс.'
) );`

etc.

In database it looks like

`INSERT INTO `wp_options` (`option_id`, `option_name`, `option_value`, `autoload`) VALUES
(183, 'gpress_options', 's:23812:\"a:394:{s:16:\"eefe9a246a783b17\";s:0:\"\";s:13:\"home_sortable\";s:201:\"a:11:{i:0;s:4:\"news\";i:1;s:7:\"service\";i:2;s:5:\"otdel\";i:3;s:5:\"stuff\";i:4;s:7:\"reasons\";i:5;s:7:\"gallery\";i:6;s:12:\"testimonials\";i:7;s:6:\"tables\";i:8;s:4:\"text\";i:9;s:9:\"attention\";i:10;s:5:\"insur\";}\";s:7:\"fav-loc\";s:1:\"1\";s:7:\"favicon\";s:2:\"14\";s:10:\"site-title\";s:1:\"2\";s:9:\"themelogo\"; ...`


Could it be related to Cyrillic symbols?

**Comment by davidsneighbour** [comment]
as long as the id's are non-cyrillic it should not be of any issue.

This database string does not look like your options though... Could it be that you have two different sets of "gpress_options" options?

**Comment by goodwinpress** [comment]
Nope. Just checked database file and theme as well, "gpress_options" used just once at sql. 

Thank you for added some markup to comment above.

**Comment by davidsneighbour** [comment]
I think issues might arise if one wp_options table has a different format than the other one. But if you transfer a full table (structure and data) this should not come up. Can you try changing the charset to one with utf8 in it? `utf8mb4_unicode_520_ci` should imho be ok. The reason you might have a Russian encoding on the table might be some sorting issues, but as far as I understand (I have only Thai language content to compare this) `utf8mb4_unicode_520_ci` will be able to sort properly by Russian alphabet (because it's part of unicode). 

I played around with your SQL line and could find it in two tables after exporting/importing. It really seems to be a special issue with your tools. 

In PhpMyAdmin you can choose the connection character set that is used when connecting to a server. You should use utf8mb4 here too.
